src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -152,19 +154,20 @@ def _accumulate_load_trades(load, grid, accumulated_trades, is_cell_tower):
     return accumulated_trades
 
 
-def _accumulate_producer_trades(load, grid, accumulated_trades):
-    accumulated_trades[load.name] = {
-        "id": load.area_id,
+def _accumulate_producer_trades(producer, grid, accumulated_trades):
+    assert producer.name not in accumulated_trades
```

Comment:
What happens if this assert is triggered? I dont see some catch, or did I miss it?

Reply:
No catch, the simulation terminates. It is deliberate, because it is a logical mistake in the code if get into this method twice for the same producer.